parlai/tasks/image_chat/agents.py

```diff
@@ -240,14 +186,111 @@ def next_example(self):
         else:
             return ready
 
-    def share(self):
+    def share(self) -> TShared:
         shared = super().share()
         shared['data'] = self.data
         shared['image_loader'] = self.image_loader
         shared['personalities'] = self.personalities
         return shared
 
 
+class GenerationTeacher(ImageChatTeacher):
+    """
+    GenerationTeacher - dialogues are split into two episodes, one from
+    each individual person's point of view.
+
+    Used in the #dodecaDialogue task. (see https://parl.ai/projects/dodecadialogue/)
+    """
+
+    def __init__(self, opt: Opt, shared: TShared = None):
+        if opt.get('datasplit', '100k') != '100k':
+            print('Warning: teacher only has 100k datasplit')
+            opt['datasplit'] = '100k'
+        if not shared:
+            self.idx_to_ep = {}
+        else:
+            self.idx_to_ep = shared['idx_to_ep']
+        self.prepend_personality = opt.get('prepend_personality', True)
+        self.include_dialogue_history = opt.get('include_dialogue_history', True)
+        super().__init__(opt, shared)
+        self.num_eps = len(self.data) + len(
```

Comment:
hmm. i don't get why you add this here? can you explain?

Reply:
yeah so in an ideal world this would be `len(self.data) * 2` since we iterate over the data twice. Unfortunately, there are a bunch of image chat episodes that are only 1 turn (i.e. just a caption) so those examples are not repeated